docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
 # Developer Guide
 
+## Table of Content
+- [Acknowledgements](#acknowledgements)
+- [Introduction](#introduction)
+- [Architecture](#architecture)
+- [Design & Implementation](#design--implementation)
+  - [Study Session Implementation](#study-session-implementation)
+  - [Timetable Implementation](#timetable-implementation)
+  - [Storage Implementation](#storage-implementation)
```

Comment:
I suggest including the sub-headings as well, eg Timer Implementation, Study session usage scenario, Design considerations for Timer class to be under Study Session Implementation.